src/components/views/DownloadModModal.vue

```diff
@@ -91,52 +83,31 @@ interface DownloadProgress {
-
-                const progressObject = {
-                    progress: 0,
-                    initialMods: [`${tsMod.getName()} (${tsVersion.getVersionNumber().toString()})`],
-                    modName: '',
-                    assignId: currentAssignId,
-                    failed: false,
-                };
-
-                store.commit('download/pushDownloadObjectToAllVersions', {
-                    assignId: currentAssignId,
-                    downloadObject: progressObject
-                });
+                const assignId = await store.dispatch(
+                    'download/addDownload',
+                    [`${tsMod.getName()} (${tsVersion.getVersionNumber().toString()})`]
+                );
 
                 setTimeout(() => {
                     ThunderstoreDownloaderProvider.instance.download(profile.asImmutableProfile(), tsMod, tsVersion, ignoreCache, (progress: number, modName: string, status: number, err: R2Error | null) => {
-                        const assignIndex = store.state.download.allVersions.findIndex(([number, val]: [number, DownloadProgress]) => number === currentAssignId);
-                        if (status === StatusEnum.FAILURE) {
-                            if (err !== null) {
-                                const existing = store.state.download.allVersions[assignIndex]
-                                existing[1].failed = true;
-                                store.commit('download/updateDownloadObject', {
-                                    assignId: assignIndex,
-                                    downloadVersion: [currentAssignId, existing[1]]
-                                });
-                                DownloadModModal.addSolutionsToError(err);
-                                return reject(err);
+                        try {
+                            if (status === StatusEnum.FAILURE) {
+                                store.commit('download/updateDownload', {assignId, failed: true});
+                                if (err !== null) {
+                                    DownloadModModal.addSolutionsToError(err);
+                                    return reject(err);
+                                }
+                            } else if (status === StatusEnum.PENDING) {
+                                store.commit('download/updateDownload', {assignId, progress, modName});
                             }
-                        } else if (status === StatusEnum.PENDING) {
-                            const obj = {
-                                progress: progress,
-                                initialMods: [`${tsMod.getName()} (${tsVersion.getVersionNumber().toString()})`],
-                                modName: modName,
-                                assignId: currentAssignId,
-                                failed: false,
+                        } catch (e) {
+                            if (e instanceof R2Error) {
```

Comment:
This should handle all caught errors regardless of the type, and in this specific method we should probably use `return reject(e)` as that's used in other places where errors are handled?